### DISCUSSION TITLE
How to use `inf` value in yaml

### DISCUSSION THREAD

**Discussion by nicolas-dufour** [facebookresearch/hydra#2480] [Q&A] [answered]
# 🐛 Bug
## Description
When writing a yaml file, inf is not interprated as a float but as an str

with

```yaml
foo: inf
```

when imported in python ```type(foo) == str```

The doc however specify that inf type is supported

## Checklist
- [ ] I checked on the latest version of Hydra
- [ ] I created a minimal repro (See [this](https://stackoverflow.com/help/minimal-reproducible-example) for tips).

## Expected Behavior

I would expect cfg.foo == float("inf")

## System information
-  Hydra 1.2
-  python 3.10
-  Linux

**Comment by Jasha10** [comment, marked as answer]
Hi @nicolas-dufour,
See Example 10.7 from [the yaml spec](http://yaml.org/spec/1.2-old/spec.html):
```yaml
infinity: !!float .inf
```
Here it is in action:
```python
>>> cfg = OmegaConf.create("foo: !!float .inf")
>>> cfg.foo
inf
```
Hydra uses [`pyyaml`](https://pyyaml.org/wiki/PyYAMLDocumentation) for parsing, and it seems that with pyyaml you can omit the `!!float` prefix:
```yaml
infinity: .inf
```
```python
>>> cfg = OmegaConf.create("foo: .inf")
>>> cfg.foo
inf
>>> import math
>>> math.isfinite(cfg.foo)
False
```

**Reply by nicolas-dufour** [reply]
Thanks! This worked out for me!

**Reply by jogepari** [reply]
Any idea how to achieve the same when overriding via hydra cli? Values overridden with .inf end up as str instead of float for me.
